src/mkfs.fat.c

```diff
@@ -1649,15 +1660,24 @@ int main(int argc, char **argv)
 	    sector_size = devinfo.sector_size;
 	    sector_size_set = 1;
 	}
+
+        if (devinfo.size <= part_offset)
+          die("The device %s size %llu is less then the offset %llu",
+              device_name, devinfo.size, (long long) part_offset);
```

Comment:
`%llu` needs `unsigned long long` type but you have passed (casted) signed `long long` type.

Reply:
Fixed